docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +99,173 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
 
-#### TestNet Release Protocol
+# neptune-core versioning and releases
 
- - Ensure that master builds against crates that live on [crates.io](https://crates.io). In particular, no dependencies on github repositories or revisions.
- - Update `README.md` in case to make sure the installation instructions are up-to-date.
+## deviation from standard semver practice
+
+neptune-core is presently both a (blockchain) library and a binary. Semantic versioning signals changes to APIs and mainly applies to libraries.  But it does not capture breaking changes that can occur in a blockchain ecosystem or breaking changes with regards to stored state, ie databases.
+
+For blockchains it is of primary importance to maintain consensus and network
+compatibility with other nodes on the network.
+
+For these reasons, we define the version identifier as follows.
+
+## version identifier
+
+neptune-core uses sem-ver compatible identifier of the form:
+```
+<major>.<minor>.<patch>[-<candidate>]
+
+where `candidate` is of the form "rc1", "rc2", etc and starts at 1.
+```
+
+Examples:
+```
+0.4.0-rc1
+0.5.0-rc1
+0.5.0-rc2
+0.5.0-rc3
+1.2.2
+```
+
+Non-Examples: (should not occur)
+```
+0.4.1-rc1       (point release should not be a release-candidate)
+1.2.0-rc0       (release candidates start at 1, not 0)
+```
+
+note: for git tags, a "v" is prefixed, as per git convention.
+
+Our release process treats these as:
+
+**major**: bumped any time there are:
+
+1. significant consensus or p2p layer changes. eg hard or soft-fork or breaking protocol change.
+2. breaking library API changes.
+3. breaking database changes.
+
+resets minor and patch to 0.
+
+**minor**: bumped any time changes from master are included in a release candidate.
+
+resets patch to 0.
+
+**patch**: bumped any time changes from a release branch are included in a release.
+
+resets candidate to "-rc1".
+
+**candidate**: added/bumped when a new public testnet is created for a candidate release.
+
+*Note that the optional suffix is supported by cargo and related tools.*
+
+## typical release
+
+The normal/typical release flow would be:
+
+1. normal changes (not consensus) are introduced in master.
+2. a release candidate is tagged from master, eg `v0.3.0-rc1` and built.
+3. a testnet is created for `v0.3.0-rc1`
+4. if significant problems are found then a new branch `v0.3.0-rc2` is created at `v0.3.0-rc1`, or possibly from master.  commit(s) are added. When ready, the final commit in the branch is tagged with `v0.3.0-rc2`, another testnet created, and so on.
+5. fixes for the release-candidate should also be applied to master, if applicable.
```

Comment:
Ah! This answers one of my questions from above. Would it make sense to add a corresponding arrow to the drawing in the intro?

Reply:
I suppose I never wrote the question down. It was essentially: (why) are fixes enabling a release not added back to the master branch?

Reply:
note added in 66b171787d18c6ad4aaf45854923545d15666d28